application/models/Logbook_model.php

```diff
@@ -1411,28 +1411,49 @@ function import_check($datetime, $callsign, $band) {
     }
   }
 
-  function lotw_update($datetime, $callsign, $band, $qsl_date, $qsl_status, $state) {
+  function lotw_update($datetime, $callsign, $band, $qsl_date, $qsl_status, $state, $qsl_gridsquare) {
 
-    if($state != "") {
-      $data = array(
-           'COL_LOTW_QSLRDATE' => $qsl_date,
-           'COL_LOTW_QSL_RCVD' => $qsl_status,
-           'COL_LOTW_QSL_SENT' => 'Y',
-           'COL_STATE' => $state
-      );
-    } else {
-      $data = array(
-           'COL_LOTW_QSLRDATE' => $qsl_date,
-           'COL_LOTW_QSL_RCVD' => $qsl_status,
-           'COL_LOTW_QSL_SENT' => 'Y'
-      );
-    }
+	$data = array(
+      'COL_LOTW_QSLRDATE' => $qsl_date,
+      'COL_LOTW_QSL_RCVD' => $qsl_status,
+      'COL_LOTW_QSL_SENT' => 'Y'
+    );
+	if($state != "") {
+      $data['COL_STATE'] = $state;
+	}
 
     $this->db->where('date_format(COL_TIME_ON, \'%Y-%m-%d %H:%i\') = "'.$datetime.'"');
     $this->db->where('COL_CALL', $callsign);
     $this->db->where('COL_BAND', $band);
 
     $this->db->update($this->config->item('table_name'), $data);
+	unset($data);
+
+	if($qsl_gridsquare != "") {
+      $data = array(
+        'COL_GRIDSQUARE' => $qsl_gridsquare
+      );
+      $this->db->where('date_format(COL_TIME_ON, \'%Y-%m-%d %H:%i\') = "'.$datetime.'"');
+      $this->db->where('COL_CALL', $callsign);
+      $this->db->where('COL_BAND', $band);
+	  $this->db->group_start();
+	  $this->db->where('COL_VUCC_GRIDS');
```

Comment:
@T012m3n7oR should this be $this->db->where('COL_VUCC_GRIDS', NULL)? COL_VUCC_GRIDS is nullable, so it makes sense to check if null or blank.

Reply:
They are one in the same.  I can change it for readability if you prefer, but codeigniter will translate that line into where `COL_VUCC_GRIDS` IS NULL

Reply:
My bad. Then I agree. I tried to check the documentation, but I didn't manage to find that it interprets it this way.

And yes, I did see the following line, that's why you use group_start and end.

No need to change it.

Reply:
I used the getCompiledUpdate function to verify it.  
To verify it you can: 
set(); the parameters, 
echo the getCompiledUpdate(<table>); back, 
and comment out the actual execute.

Because I do not have a separate test environment getCompiledUpdate lets me test without risking my data.

Reply:
@T012m3n7oR thanks. I'm still learning Codeigniter stuff. I usually find it easier to write out the sql (at least more complicated quiers) as a string and test it out in DBeaver (which is connected to my MariaDB).